src/odb.c

```diff
@@ -1521,6 +1521,25 @@ void git_odb_backend_data_free(git_odb_backend *backend, void *data)
 	git__free(data);
 }
 
+int git_odb_set_backend_priority(git_odb *db, size_t index, int priority)
+{
+	size_t i;
+	assert(db);
+
+	if (index > db->backends.length - 1) {
```

Comment:
When `db->backends.length == 0`, this subtraction will make it `SIZE_MAX` and out-of-bounds `index` values will be let through.

Reply:
@ethomson good point! I addressed this in a new commit by asserting the db->backends.length.

Reply:
🤦 My previous fix won't work for release builds. Updated with more logic based on fetching the object from the vector. Thanks again!